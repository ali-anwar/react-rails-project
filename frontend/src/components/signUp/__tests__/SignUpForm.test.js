import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {SignUpForm} from '../SignUpForm';



function setup(heading) {
  const props = {
    handleSubmit: jest.fn(),
    pristine: true,
    reset: jest.fn(),
    submitting: false,
    heading: heading,
    authors: [],
    handleSave: jest.fn(),
    handleSubmit: jest.fn(),
    roles: ['User', 'Admin']
  };

  return shallow(<SignUpForm {...props}/>);
}

describe('SignUpForm.test.js', () => {

  it('renders without crashing', () => {
      const wrapper = setup('Add');
      expect(wrapper.length).toEqual(1);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
  });


  it('renders form and display "Add" in h1', () => {
      const wrapper = setup('Add');
      expect(wrapper.find('h1').text()).toEqual('Add');
  });


  it('displays "Edit" in h1', () => {
      const wrapper = setup('Edit');
      expect(wrapper.find('h1').text()).toEqual('Edit');
  });
});

