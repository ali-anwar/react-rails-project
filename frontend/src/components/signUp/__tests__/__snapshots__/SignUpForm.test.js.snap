exports[`SignUpForm.test.js renders without crashing 1`] = `
<form>
  <div
    className="mt-4 mb-4">
    <h1>
      Add
    </h1>
  </div>
  <Field
    component={[Function]}
    label="Name"
    name="name"
    type="text" />
  <Field
    component={[Function]}
    label="Email"
    name="email"
    type="email" />
  <Field
    component={[Function]}
    label="Password"
    name="password"
    type="password" />
  <Field
    component={[Function]}
    label="Password Confirmation"
    name="confirmpassword"
    type="password" />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit">
      Sign Up
    </button>
  </div>
</form>
`;
